.github/workflows/nightly-no-cache.yaml

```diff
@@ -0,0 +1,110 @@
+name: Nightly build
+
+on:
+  schedule:
+    # strange schedule to reduce the risk of DDOS GitHub infra
+    - cron: "48 4 * * *"
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set up JDK 11 for running checks
+        uses: actions/setup-java@v2
+        with:
+          distribution: adopt
+          java-version: 11
+
+
+      - name: Build
+        run: ./gradlew build --stacktrace -x :smoke-tests:test --no-build-cache
```

Comment:
Just curious is it possible to use top level `env` and `GRADLE_OPTS` for some DRY?